common/djangoapps/track/backends/tests/test_oli.py

```diff
@@ -0,0 +1,247 @@
+"""Tests OLIAnalyticsBackend: Transmit events to the OLI analytics service"""
+from __future__ import absolute_import
+
+from mock import patch, Mock
+
+from django.test import TestCase
+from south.utils import datetime_utils as datetime
+
+from track.backends.oli import OLIAnalyticsBackend
+from student.tests.factories import UserFactory
+
+
+# pylint: disable=line-too-long
+class TestOLIBackend(TestCase):
+    """
+    Tests OLIAnalyticsBackend.  OLI should only process 'problem_check' event type.
+    Course IDs must be part of the whitelisted set
+    """
+    def setUp(self):
+        self.patcher = patch('track.backends.oli.OAuth1Session.put', mock_put)
+        self.addCleanup(self.patcher.stop)
+        self.patcher.start()
+        self.backend = OLIAnalyticsBackend()
+
+        # create a test user
+        self.user = UserFactory.create()
+        self.user.save()
+
+        # initialize attributes for the test cases.
+        self.backend.url = 'https://example.com'
+        self.backend.path = 'path'
+        self.backend.key = 'key'
+        self.backend.secret = 'secret'
+        self.backend.course_id_set = ['edX/DemoX/Demo_Course', 'edX/DemoX/Demo_Course1']
+
+    def test_oli_backend_correct_event_type(self):
+
+        # OLI only cares about events with event_type = 'problem_check'
+        # This is a sample problem_check event
+
+        event = {
+            'username': 'verified',
+            'event_type': 'problem_check',
+            'ip': '10.0.2.2',
+            'agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.124 Safari/537.36',
```

Comment:
Is there anything special here or is this just your local user agent?

It may be worth a clarifying comment. Often when I see a line like this, it makes me think there's a specific reason why such a particular string was chosen. If I can freely change this value, it'd be nice to know that the value is arbitrarily chosen.